File: app/src/main/java/com/jobtick/activities/AddTagActivity.java
package com.jobtick.activities;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.appbar.MaterialToolbar;
import com.jobtick.edit_text.EditTextRegular;
import com.jobtick.R;
import com.jobtick.TextView.TextViewSemiBold;
import com.jobtick.adapers.AddTagAdapter;
import com.jobtick.utils.ConstantKey;
import com.jobtick.utils.Tools;
import com.jobtick.widget.SpacingItemDecoration;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class AddTagActivity extends ActivityBase {

    @BindView(R.id.toolbar)
    MaterialToolbar toolbar;
    @BindView(R.id.recycler_view)
    RecyclerView recyclerView;
    AddTagAdapter adapter;
    ArrayList<String> addTagList;
    @BindView(R.id.txt_title)
    TextViewSemiBold txtTitle;
    @BindView(R.id.edt_add_tag)
    EditTextRegular edtAddTag;
    @BindView(R.id.img_btn_add_tag)
    ImageView imgBtnAddTag;
    String action_bar_title;
    String title;
    private int tag_array_size = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_tag);
        ButterKnife.bind(this);
        addTagList = new ArrayList<>();
        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            if (bundle.getStringArrayList(ConstantKey.TAG) != null) {
                addTagList = bundle.getStringArrayList(ConstantKey.TAG);
            }
            if (bundle.getString(ConstantKey.ACTIONBAR_TITLE) != null) {
                action_bar_title = bundle.getString(ConstantKey.ACTIONBAR_TITLE);
            }
            if (bundle.getString(ConstantKey.TITLE) != null) {
                title = bundle.getString(ConstantKey.TITLE);
            }
            if (bundle.getInt(ConstantKey.TAG_SIZE) != 0) {
                tag_array_size = bundle.getInt(ConstantKey.TAG_SIZE);
            }
        }
        init();
        initToolbar();
    }

    private void initToolbar() {
        toolbar.setNavigationOnClickListener(v -> onBackPressed());
        toolbar.setTitle(action_bar_title);
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent();
        intent.putExtra(ConstantKey.TAG, addTagList);
        setResult(25, intent);
        super.onBackPressed();

    }


    private void init() {
        txtTitle.setText(title);
        recyclerView.setLayoutManager(new LinearLayoutManager(AddTagActivity.this));
        recyclerView.addItemDecoration(new SpacingItemDecoration(1, Tools.dpToPx(AddTagActivity.this, 5), true));
        recyclerView.setHasFixedSize(true);

        adapter = new AddTagAdapter(addTagList, data -> {
            addTagList.remove(data);
            adapter.updateItem(addTagList);
            recyclerView.swapAdapter(adapter, true);
        });
        recyclerView.setAdapter(adapter);
    }


    @OnClick(R.id.img_btn_add_tag)
    public void onViewClicked() {
        if (TextUtils.isEmpty(edtAddTag.getText().toString().trim())) {
            edtAddTag.setError("Text is empty");
            return;
        } else {
            if (tag_array_size > addTagList.size()) {
                if (recyclerView.getVisibility() != View.VISIBLE) {
                    recyclerView.setVisibility(View.VISIBLE);
                }
                addTagList.add(edtAddTag.getText().toString().trim());
                adapter.notifyItemInserted(adapter.getItemCount());
                edtAddTag.setText(null);
            } else {
                showToast("Max. 3 Tag you can add", AddTagActivity.this);
            }
        }
    }
}

File: app/src/main/java/com/jobtick/android/activities/AddTagActivity.kt
package com.jobtick.android.activities

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.appbar.MaterialToolbar
import com.jobtick.android.R
import com.jobtick.android.adapers.AddTagAdapter
import com.jobtick.android.edittext.EditTextRegular
import com.jobtick.android.utils.ConstantKey
import com.jobtick.android.utils.Tools
import com.jobtick.android.widget.SpacingItemDecoration
import java.util.*

class AddTagActivity : ActivityBase() {
    private var toolbar: MaterialToolbar? = null
    private var recyclerView: RecyclerView? = null
    private var adapter: AddTagAdapter? = null
    private var addTagList: ArrayList<String>? = null
    private var txtTitle: TextView? = null
    private var edtAddTag: EditTextRegular? = null
    private var imgBtnAddTag: ImageView? = null
    private var actionBarTitle: String? = null
    private var title: String? = null
    private var tagArraySize = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_tag)
        setIDs()
        addTagList = ArrayList()
        val bundle = intent.extras
        if (bundle != null) {
            if (bundle.getStringArrayList(ConstantKey.TAG) != null) {
                addTagList = bundle.getStringArrayList(ConstantKey.TAG)
            }
            if (bundle.getString(ConstantKey.ACTIONBAR_TITLE) != null) {
                actionBarTitle = bundle.getString(ConstantKey.ACTIONBAR_TITLE)
            }
            if (bundle.getString(ConstantKey.TITLE) != null) {
                title = bundle.getString(ConstantKey.TITLE)
            }
            if (bundle.getInt(ConstantKey.TAG_SIZE) != 0) {
                tagArraySize = bundle.getInt(ConstantKey.TAG_SIZE)
            }
        }
        init()
        initToolbar()
    }

    private fun setIDs() {
        toolbar  = findViewById(R.id.toolbar)
        recyclerView  = findViewById(R.id.recycler_view)
        txtTitle  = findViewById(R.id.txt_title)
        edtAddTag  = findViewById(R.id.edt_add_tag)
        imgBtnAddTag  = findViewById(R.id.img_btn_add_tag)
        onViewClick()
    }

    private fun initToolbar() {
        toolbar!!.setNavigationOnClickListener { v: View? -> onBackPressed() }
        toolbar!!.title = actionBarTitle
    }

    override fun onBackPressed() {
        val intent = Intent()
        intent.putExtra(ConstantKey.TAG, addTagList)
        setResult(25, intent)
        super.onBackPressed()
    }

    private fun init() {
        edtAddTag!!.hint = title
        recyclerView!!.layoutManager = LinearLayoutManager(this@AddTagActivity)
        recyclerView!!.addItemDecoration(SpacingItemDecoration(1, Tools.dpToPx(this@AddTagActivity, 5), true))
        recyclerView!!.setHasFixedSize(true)
        adapter = AddTagAdapter(addTagList, true) { data: String? ->
            addTagList!!.remove(data)
            adapter!!.updateItem(addTagList)
            recyclerView!!.swapAdapter(adapter, true)
        }
        recyclerView!!.adapter = adapter
    }

    private fun onViewClick() {
        imgBtnAddTag!!.setOnClickListener {
            if (TextUtils.isEmpty(edtAddTag!!.text.toString().trim { it <= ' ' })) {
                edtAddTag!!.error = "Text is empty"
                return@setOnClickListener
            } else {
                if (tagArraySize > addTagList!!.size) {
                    if (recyclerView!!.visibility != View.VISIBLE) {
                        recyclerView!!.visibility = View.VISIBLE
                    }
                    addTagList!!.add(edtAddTag!!.text.toString().trim { it <= ' ' })
                    adapter!!.notifyItemInserted(adapter!!.itemCount)
                    edtAddTag!!.text = null
                } else {
                    showToast("Max. 3 Tag you can add", this@AddTagActivity)
                }
            }
        }
    }
}